cli/running/running_test.go

```diff
@@ -113,20 +113,87 @@ func Test_CollectInstances(t *testing.T) {
 	instances, err = CollectInstances("script", instancesEnabledPath,
 		integrity.IntegrityCtx{
 			Repository: &mockRepository{},
-		}, true)
+		}, ConfigLoadAll)
 	assert.NoError(t, err)
 	assert.Equal(t, 1, len(instances))
 
 	require.NoError(t, os.Chmod(instancesEnabledPath, 0666))
 	instances, err = CollectInstances("script", instancesEnabledPath,
 		integrity.IntegrityCtx{
 			Repository: &mockRepository{},
-		}, true)
+		}, ConfigLoadAll)
 	assert.ErrorContains(t, err, "script.lua: permission denied")
 	assert.Equal(t, 1, len(instances))
 	require.NoError(t, os.Chmod(instancesEnabledPath, 0755))
 }
 
+func Test_CollectInstancesInstanceScript(t *testing.T) {
+	if user, err := user.Current(); err == nil && user.Uid == "0" {
+		t.Skip("Skipping the test, it shouldn't run as root")
+	}
+	tmpDir := t.TempDir()
+	instancesEnabledPath := filepath.Join(tmpDir, "instances.enabled")
+	require.NoError(t, os.Mkdir(instancesEnabledPath, 0755))
+
+	err := os.WriteFile(filepath.Join(instancesEnabledPath, "script.lua"),
+		[]byte("print(42)"), 0644)
+	require.NoError(t, err)
+
+	instances, err := CollectInstances("script", instancesEnabledPath,
+		integrity.IntegrityCtx{
+			Repository: &mockRepository{},
+		}, ConfigLoadSkip)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(instances))
+
+	instances, err = CollectInstances("script", instancesEnabledPath,
+		integrity.IntegrityCtx{
+			Repository: &mockRepository{},
+		}, ConfigLoadCluster)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(instances))
+
+	instances, err = CollectInstances("script", instancesEnabledPath,
+		integrity.IntegrityCtx{
+			Repository: &mockRepository{},
+		}, ConfigLoadAll)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(instances))
+
+	require.NoError(t, os.Chmod(instancesEnabledPath, 0666))
+	_, err = CollectInstances("script", instancesEnabledPath,
+		integrity.IntegrityCtx{
+			Repository: &mockRepository{},
+		}, ConfigLoadAll)
+	assert.ErrorContains(t, err, "script.lua: permission denied")
+	require.NoError(t, os.Chmod(instancesEnabledPath, 0755))
+}
+
+func Test_CollectInstancesEtcdNotAvailable(t *testing.T) {
+	if user, err := user.Current(); err == nil && user.Uid == "0" {
+		t.Skip("Skipping the test, it shouldn't run as root")
+	}
+	instancesEnabledPath := filepath.Join("testdata", "instances_enabled")
+
+	_, err := CollectInstances("config_load", instancesEnabledPath,
+		integrity.IntegrityCtx{
+			Repository: &mockRepository{},
+		}, ConfigLoadAll)
+	assert.ErrorContains(t, err, "unable to connect to etcd")
+
+	_, err = CollectInstances("config_load", instancesEnabledPath,
+		integrity.IntegrityCtx{
+			Repository: &mockRepository{},
+		}, ConfigLoadCluster)
+	assert.ErrorContains(t, err, "unable to connect to etcd")
+
+	_, err = CollectInstances("config_load", instancesEnabledPath,
+		integrity.IntegrityCtx{
+			Repository: &mockRepository{},
+		}, ConfigLoadSkip)
+	assert.NoError(t, err)
```

Comment:
Such tests are better to test with a table tests approach: https://go.dev/wiki/TableDrivenTests
```suggestion
	cases := []struct {
		mode ConfigLoad
		err      string
	} {
		{
			mode: ConfigLoadAll,
			err:     "unable to connect to etcd",
		},
		{
			mode: ConfigLoadCluster,
			err:     "unable to connect to etcd",
		},
		{
			mode: ConfigLoadSkip,
		},
	}

	for _, tc := cases {
		t.Run(t.err, func(t *testing.T) {
			_, err := CollectInstances("config_load", instancesEnabledPath,
				integrity.IntegrityCtx{
					Repository: &mockRepository{},
				}, tc.mode)
			if tc.err != "" {
				assert.ErrorContains(t, err, tc.err)
			} else {
				assert.NoError(t, err)
			}
		})
	}
```